File: qqbot-admin-api/src/main/java/com/mieai/qqbot/admin/events/PluginDeliveryQueueStatsResponse.java
package com.mieai.qqbot.admin.events;

import com.mieai.qqbot.persistence.plugin.PluginDeliveryQueueStats;

public record PluginDeliveryQueueStatsResponse(
        long totalCount,
        long pendingCount,
        long inProgressCount,
        long retryWaitCount,
        long succeededCount,
        long deadLetterCount,
        long pausedCount) {

    static PluginDeliveryQueueStatsResponse from(PluginDeliveryQueueStats stats) {
        return new PluginDeliveryQueueStatsResponse(
                stats.totalCount(), stats.pendingCount(), stats.inProgressCount(),
                stats.retryWaitCount(), stats.succeededCount(), stats.deadLetterCount(),
                stats.pausedCount());
    }
}

File: qqbot-module-operations/src/main/kotlin/com/mieai/qqbot/admin/events/PluginDeliveryQueueStatsResponse.kt
package com.mieai.qqbot.admin.events

import com.mieai.qqbot.persistence.plugin.PluginDeliveryQueueStats

data class PluginDeliveryQueueStatsResponse(
    val totalCount: Long,
    val pendingCount: Long,
    val inProgressCount: Long,
    val retryWaitCount: Long,
    val succeededCount: Long,
    val deadLetterCount: Long,
    val pausedCount: Long,
) {
    companion object {
        fun from(stats: PluginDeliveryQueueStats): PluginDeliveryQueueStatsResponse =
            PluginDeliveryQueueStatsResponse(
                stats.totalCount,
                stats.pendingCount,
                stats.inProgressCount,
                stats.retryWaitCount,
                stats.succeededCount,
                stats.deadLetterCount,
                stats.pausedCount,
            )
    }
}
